source/common/http/http1/codec_impl.cc

```diff
@@ -493,6 +499,9 @@ Status ConnectionImpl::completeLastHeader() {
   ENVOY_CONN_LOG(trace, "completed header: key={} value={}", connection_,
                  current_header_field_.getStringView(), current_header_value_.getStringView());
 
+  // saving the original case-preserved header field here.
+  std::string current_header_field_orig(std::string(current_header_field_.getStringView()));
```

Comment:
I think if/when you change the logic to check preserved case before even setting the key/value on the map, you'll be able to remove this copy (that is unnecessary for all but the ones that opt in to preserving case)